nupic/research/frameworks/vernon/mixins/oml.py

```diff
@@ -103,13 +102,15 @@ def find_best_lr(self, num_classes_learned):
                     self.reset_params(params)
 
                 optim = Adam(params, lr=lr),
-                train_model(
-                    model=self.model.module,
-                    loader=self.test_train_loader,
-                    optimizer=optim,
-                    device=self.device,
-                    criterion=self._loss_function,
-                )
+                for task in new_tasks:
```

Comment:
This seems to be more accurate. But just as a sanity check, is this how the OML repo is doing? Because if I understood correctly from the first PR the whole meta testing phase was a copy and paste from OML with the few adaptations required to match vernon structure.

Reply:
It was copy-paste, but we had to translate the parts that dealt with iterating through the data. The OML repo didn't iterate task-by-task because they pre-ordered their dataset to iterate over the tasks in sequence (See this [line](https://github.com/khurramjaved96/mrcl/blob/2855a6b7e820f171432981b58c49664fcdbf00ed/evaluate_omniglot.py#L144)). In contrast, our samplers automatically shuffle the data while we iterate through the dataloader.

Note, in the future, I think it will be better to make sure our task samplers (such as `TaskDistributedSampler`) can preserve the order of the tasks. They technically can if you set `shuffle=False`, but I wondered about another case where we preserve the order of the tasks but then shuffle samples within each task. I left this till later to keep things simple for now.

Reply:
That is a good observation. We definitely can distinguish between shuffling tasks and shuffling samples within tasks in the sampler.